packages/remediations/src/NewRemediationWizard.js

```diff
@@ -137,33 +140,54 @@ class RemediationWizard extends Component {
-                        setHideForm: this.setHideForm }}>
-                    <FormRenderer
-                        schema={this.state.schema}
-                        container={this.container}
-                        subscription={{ values: true }}
-                        FormTemplate={this.getFormTemplate}
-                        initialValues={{
-                            multiple: !!this.state.resolutions?.find(r => r.resolutions.length > 1),
-                            'manual-resolution': true,
-                            'selected-resolutions': {}
-                        }}
-                        componentMapper={{
-                            [componentTypes.WIZARD]: Wizard,
-                            ...this.state.mapperExtension
-                        }}
-                        onSubmit={() => undefined}
-                        onCancel={this.closeWizard}
-                    />
-                </RemediationWizardContext.Provider>
+                <FormRenderer
+                    schema={this.state.schema}
+                    container={this.container}
+                    subscription={{ values: true }}
+                    FormTemplate={this.getFormTemplate}
+                    initialValues={{
+                        multiple: !!this.state.resolutions?.find(r => r.resolutions.length > 1),
+                        'manual-resolution': true,
+                        'selected-resolutions': {}
+                    }}
+                    componentMapper={{
+                        [componentTypes.WIZARD]: Wizard,
+                        ...this.state.mapperExtension
+                    }}
+                    onSubmit={(_, formOptions) => {
+                        const values = formOptions.getState().values;
+                        const issues = this.state.data.issues.map(({ id }) => ({
+                            id,
+                            resolution: values['manual-resolution'] ? this.getResolution(id)?.[0]?.id : undefined,
+                            systems: this.state.data.systems
+                        }));
+                        const add = { issues, systems: this.state.data.systems };
+                        if (values['existing-playbook-selected']) {
+                            const { id, name } = values['existing-playbook'];
+                            // eslint-disable-next-line camelcase
+                            api.patchRemediation(id, { add, auto_reboot: values['auto-reboot'] }, this.state.basePath)
+                            .then(() => this.resolver(this.state.deferred)(id, name, false));
+                        } else {
+                            // eslint-disable-next-line camelcase
+                            api.createRemediation({ name: values['select-playbook'], add, auto_reboot: values['auto-reboot'] }, this.state.basePath)
+                            .then(({ id }) => this.resolver(this.state.deferred)(id, values['select-playbook'], true));
```

Comment:
I would also create constants for values like `select-playbook`. It will make the code bit safer and we can reuse them in the schema as well.